sycl/source/detail/device_image_impl.hpp

```diff
@@ -95,10 +95,14 @@ class device_image_impl {
 
     // Otherwise, if the device candidate is a sub-device it is also valid if
     // its parent is valid.
-    if (!getSyclObjImpl(DeviceCand)->isRootDevice())
-      return has_kernel(KernelIDCand,
-                        DeviceCand.get_info<info::device::parent_device>());
-
+    if (!getSyclObjImpl(DeviceCand)->isRootDevice()) {
+      try {
+        return has_kernel(KernelIDCand,
+                          DeviceCand.get_info<info::device::parent_device>());
+      } catch (std::exception &e) {
```

Comment:
Just to double-check, `std::exception` or `sycl::exception`?

Reply:
std::exception is correct.
sycl::exception is an extension of std::exception.
At this outermost level, we have to catch whatever propagated from the origin, and exit gracefully with some details of the exception.